### PR TITLE
fix: Remove -Xlint:unchecked, suppress all existing violations, add @CanIgnoreReturnValue

### LINKED ISSUE #1323: unchecked casts are bad practice, but allowed globally in pom.xml
The pom.xml adds `-Xlint:unchecked` globally. We should remove this, and deal with existing violations individually.

### PR DESCRIPTION
**Pull request by sinelaw**
Fixes #1323  ☕️

This will prevent further violations from being added without explicit suppressions.

A future step would be to review all existing violations and fix instead of suppressing them.

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java [hunk 1]
```diff
@@ -418,7 +419,7 @@ public boolean createScopedRequired() {
   @Override
   public GoogleCredentials createScoped(Collection<String> scopes) {
     return toBuilder()
-        .setScopes(new ArrayList(scopes))
+        .setScopes(new ArrayList<>(scopes))
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java:422]: why this?

Comment (sinelaw) [reply oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java:422]: To avoid using raw types (ArrayList vs. generic ArrayList<>)

**Comment by TimurSadykov** [comment]
@sinelaw thanks for the contribution, it makes sense, please see a single nit comment

**Comment by sinelaw** [comment]
Added another commit - PTAL

**Review by TimurSadykov — APPROVED** [review]
LGTM

**Pull request merged**